NitroxServer-Subnautica/GameLogic/Entities/Spawning/SubnauticaEntitySpawnPointFactory.cs

```diff
@@ -20,21 +20,23 @@ public override List<EntitySpawnPoint> From(AbsoluteEntityCell absoluteEntityCel
                     List<EntitySlot.Type> slotTypes = SlotsHelper.GetEntitySlotTypes(entitySlotData);
                     List<string> stringSlotTypes = slotTypes.Select(s => s.ToString()).ToList();
 
-                    spawnPoints.Add(
-                        new EntitySpawnPoint(absoluteEntityCell,
-                                             entitySlotData.localPosition,
-                                             entitySlotData.localRotation,
-                                             stringSlotTypes,
-                                             entitySlotData.density,
-                                             entitySlotData.biomeType.ToString())
-                    );
+                    new EntitySpawnPoint(absoluteEntityCell,
+                                         entitySlotData.localPosition,
+                                         entitySlotData.localRotation,
+                                         stringSlotTypes,
+                                         entitySlotData.density,
+                                         entitySlotData.biomeType.ToString(),
+                                         gameObject.Id,
+                                         gameObject.Parent);
                 }
             }
             else
             {
-                spawnPoints.Add(
-                    new EntitySpawnPoint(absoluteEntityCell, transform.Position, transform.Rotation, transform.Scale, gameObject.ClassId)
-                );
+                EntitySpawnPoint spawnPoint = new EntitySpawnPoint(absoluteEntityCell, transform.Position, transform.Rotation, transform.Scale, gameObject.ClassId, gameObject.Id, gameObject.Parent);
+                if (spawnPoint.Parent == null)
```

Comment:
We only add the spawn point if it has a parent?

Reply:
If the spawnPoint has no parent it will be added to the list of entities to spawn(aka its the root parent of other entities) and other entities will be setting their parent as that entity.(gameObject.Parent = gameObject.ParentId)